components/app/Calendar.vue

```diff
@@ -2,18 +2,43 @@
 import VueCal from 'vue-cal'
 import 'vue-cal/dist/vuecal.css'
 
-defineProps<{
+const props = defineProps<{
   group: any
   pending: boolean
 }>()
 
+const isMobile = useMediaQuery('(max-width: 768px)')
+const calendarView = useCookie('calendarView', { default: () => 'calendar' })
+const selectedView = computed({
+  get: () => isMobile.value ? 'list' : calendarView.value,
+  set: value => calendarView.value = value,
+})
 const showEventModal = ref(false)
 const selectedEvent = ref<any>({})
 const onEventClick = (event: any, e: any) => {
   selectedEvent.value = event
   showEventModal.value = true
   e.stopPropagation()
 }
+
+// used to display events grouped by month in the list view
+const groupedEvents = computed(() => {
+  return props.group.items.reduce((acc: any, event: any) => {
+    // creating a key to group the events by month in format "Month Year"
+    const monthKey = event.start.toLocaleDateString('en-US', { year: 'numeric', month: 'long' })
+
+    // check if there are already events in this month
+    // if not, initialize an empty array for this month
+    if (!acc[monthKey]) {
+      acc[monthKey] = []
+    }
+
+    // add current in the same month to the array
+    acc[monthKey].push(event)
+
+    return acc
+  }, {})
+})
```

Comment:
Awesome I like the use of using a computed value to group the events. I personally might change the `acc` on line 26 to something like `monthEvents` but not a show stopper whatsoever.